### PR TITLE
04 auth

### PR DESCRIPTION
**Pull request by YuliaNikolaeva**

### CODE REVIEW

File: api/auth/auth.controller.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+        const findUserByMail = await UserModel.findOne({email});
+
+        if(findUserByMail) {
+            return res.status(409).send({
+                message: "Email in use"
+            });
+        };
+
+        const hashedPass = await bcrypt.hash(body.password, Number(process.env.SALT));
```
Comment (odykyi) [inline api/auth/auth.controller.js:24]: validate if password provided

File: api/auth/auth.controller.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+const loginController = async (req, res, next) => {
+    try {
+        const {body} = req;
+        const {email, password} = body;
+
+        const foundUser = await UserModel.findOne({email});
+
+        if(!foundUser) {
+            return res.status(404).json(
+                {"message": "Email or password is wrong"}
```
Comment (odykyi) [inline api/auth/auth.controller.js:51]: User not found

File: api/auth/auth.controller.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+        if(!isPassEqual) {
+            return res.status(401).json(
+                {"message": "Email or password is wrong"}
+            );
+        };
+
+        const accessToken = await createVerificationToken({id: foundUser._id});
+
+        const addToken = await UserModel.findByIdAndUpdate(foundUser._id, {token: accessToken}, {new: true})
```
Comment (odykyi) [inline api/auth/auth.controller.js:65]: variable addToken  not used

File: contacts.js [hunk 4]
```diff
@@ -58,7 +58,7 @@ function addContact(name, email, phone) {
             phone,
         };
         const newArrAllContacts = [...JSON.parse(data), newContact];
-      
+
```
Comment (odykyi) [inline contacts.js:61]: move all CRUD logic from json file to mongodb

File: api/auth/auth.controller.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+        const currentUser = await UserModel.findById(foundUserByEmail._id);
+
+        if(!currentUser.token) {
+            return res.status(401).json(
+                {"message": "Not authorized"}
+            );
+        };
+
+       const deleteToken = await UserModel.findByIdAndUpdate(currentUser._id, {token: ''}, {new: true})
```
Comment (odykyi) [inline api/auth/auth.controller.js:112]: deleteToken variable not used

**Review by odykyi — CHANGES_REQUESTED** [review]
comments

**Comment by YuliaNikolaeva** [comment]
created new pull request

**Pull request closed without merging**